driver-core/src/main/com/mongodb/connection/netty/ReadTimeoutHandler.java

```diff
@@ -40,13 +40,16 @@ public ReadTimeoutHandler(final long readTimeout) {
     }
 
     void scheduleTimeout(final ChannelHandlerContext ctx) {
-        timeout = ctx.executor().schedule(new ReadTimeoutTask(ctx), readTimeout, TimeUnit.MILLISECONDS);
+        if (timeout == null) {
```

Comment:
I think it should be an invariant that a timeout should not be scheduled unless the previous one has been removed.  So if timeout != null, should throw instead of ignoring.